test/Dialect/Tpp/tpp-bufferization.mlir

```diff
@@ -590,3 +590,35 @@ func.func @fused_brgemm_op(%arg0: tensor<1x3x3xf32>, %arg1: tensor<1x3x3xf32>,
      %arg2: tensor<3x3xf32>, %arg3: tensor<3x3xf32>) -> tensor<3x3xf32>
   return %0: tensor<3x3xf32>
 } 
+
+// -----
+
+func.func @conflict_in_place(%arg0: tensor<3x3xf32>) -> tensor<3x3xf32> {
+  %0 = tpp.add(%arg0: tensor<3x3xf32>, %arg0: tensor<3x3xf32>) -> tensor<3x3xf32>
+  %1 = tpp.add(%0: tensor<3x3xf32>, %arg0: tensor<3x3xf32>) -> tensor<3x3xf32>
+  return %1 : tensor<3x3xf32>
+}
+
+// CHECK-LABEL: conflict_in_place
+// CHECK-SAME: %[[ARG:.+]]: memref<3x3xf32>
+// CHECK: %[[ALLOC:.+]] = memref.alloc() {{.+}} : memref<3x3xf32>
```

Comment:
Is this extra copy because you use `%arg0` twice?

Can you have a test that uses a local constant to make sure there's no extra alloc?

Reply:
Yes, the second ops uses %arg0, so %arg0 is not writable for the first op.